attachment_zipped_download/models/ir_attachment.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2019 César Fernández Domínguez <cesfernandez@outlook.com>
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import zipfile
+from io import BytesIO
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    def action_attachments_download(self):
+        items = self.filtered(lambda x: x.type == "binary")
+        if not items:
+            raise UserError(
+                _("None attachment selected. Only binary attachments allowed.")
+            )
+        ids = ",".join(map(str, items.ids))
+        return {
+            "type": "ir.actions.act_url",
+            "url": "/web/attachment/download_zip?ids=%s" % (ids),
+            "target": "self",
+        }
+
+    def _create_temp_zip(self):
+        zip_buffer = BytesIO()
+        with zipfile.ZipFile(zip_buffer, "a", zipfile.ZIP_DEFLATED, False) as zip_file:
```

Comment:
This means no compression, isn't it? Put on ROADMAP to make this configurable.

Reply:
Not exactly, the explanation from the https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile documentation is as follows:

_compression is the ZIP compression method to use when writing the archive, and should be ZIP_STORED, ZIP_DEFLATED, ZIP_BZIP2 or ZIP_LZMA; unrecognized values will cause NotImplementedError to be raised. If ZIP_DEFLATED, ZIP_BZIP2 or ZIP_LZMA is specified but the corresponding module (zlib, bz2 or lzma) is not available, RuntimeError is raised. The default is ZIP_STORED._